sys-devel/llvm/llvm-18.1.2.ebuild

```diff
@@ -340,6 +350,52 @@ get_distribution_components() {
 		use debuginfod && out+=(
 			llvm-debuginfod
 		)
+		use polly && out+=(
+			Polly
+			PollyISL
+			LLVMAggressiveInstCombine
```

Comment:
If it means installing all static libraries again, that's a big "no".

Reply:
The issue here is that if I remove those lines I get the following hint in configure stage:

 ```* get_distribution_components() is outdated!
 * Add: Polly PollyISL llvm-debuginfod
 * Remove:
```

But adding those without its deps makes cmake not successful so this is why it is a large list.
Furthermore, LLVMExtensions needs to be installed or otherwise clang (and possibly other frontends) will not compile.

Is there any better aproach?

Reply:
This is the error that clang gives in the case of installing llvm+polly without listing static libraries explicitely in `get_distribution_components`:

```
CMake Error at /usr/lib/llvm/18/lib/cmake/llvm/AddLLVM.cmake:1131 (set_property):
  set_property could not find TARGET LLVMExtensions.  Perhaps it has not yet
  been created.
Call Stack (most recent call first):
  CMakeLists.txt:922 (process_llvm_pass_plugins)


CMake Error at /usr/lib/llvm/18/lib/cmake/llvm/AddLLVM.cmake:1132 (set_property):
  set_property could not find TARGET LLVMExtensions.  Perhaps it has not yet
  been created.
Call Stack (most recent call first):
  CMakeLists.txt:922 (process_llvm_pass_plugins)
```

Reply:
If they don't need to be installed, you can just add them to the exclude cases in `check_distribution_components`.

Reply:
> If they don't need to be installed, you can just add them to the exclude cases in `check_distribution_components`.

Thanks for the reply, unfortunatelly I am not sure if I understood you correctly as the only way I found without get_distribution_component outdated warning and making cmake to pass configure and build is the current way:

* Reducing the out+=( list in `get_distribution_components` causes cmake errors on EXPORTs
* Reducing the cases on `check_distribution_components` (letting the rest fall into LLVM* and being ignored) makes warnings appear to hint add/remove items and the libraries being installed anyway (seems they are dep by cmake)

Is possible that they really *need* to be installed as dependencies?
If not, then I did not understood you correctly and I tried everything I imagined, so can you please give a quick test when you find time and tell me one concrete hint so I can continue and finish for a better review?

Thanks

Reply:
Don't "reduce" `check_distribution_components`, extend it to list Polly targets.

Reply:
Isn't that what I already did at line 143 (pasted below for reference)? It may seem large but that's what polly needs if built statically.
```
					# Polly + deps
					Polly|LLVMAggressiveInstCombine|LLVMAnalysis|LLVMAsmParser|LLVMBinaryFormat|LLVMBitReader|LLVMBitWriter|LLVMBitstreamReader|LLVMCFGuard|LLVMCodeGen|LLVMCodeGenTypes|LLVMCore|LLVMCoroutines|LLVMDebugInfoBTF|LLVMDebugInfoCodeView|LLVMDebugInfoDWARF|LLVMDebugInfoMSF|LLVMDebugInfoPDB|LLVMExtensions|LLVMFrontendOffloading|LLVMFrontendOpenMP|LLVMHipStdPar|LLVMIRPrinter|LLVMIRReader|LLVMInstCombine|LLVMInstrumentation|LLVMLinker|LLVMMC|LLVMMCParser|LLVMObjCARCOpts|LLVMObject|LLVMPasses|LLVMProfileData|LLVMRemarks|LLVMScalarOpts|LLVMSymbolize|LLVMTarget|LLVMTargetParser|LLVMTextAPI|LLVMTransformUtils|LLVMVectorize|LLVMipo)
						use polly || continue
						;;
```

Reply:
Sigh. Let's approach this differently.

Stop installing `.a` files, ignore QA warnings. I'll handle them later.

Also, please limit the change to 19.0.0.9999, I'll backport it from there.